src/dbally/iql_generator/prompt.py

```diff
@@ -71,82 +112,131 @@ def __init__(self, *, question: str, examples: List[FewShotExample] = None) -> N
 
 class IQLGenerationPromptFormat(PromptFormat):
     """
-    IQL prompt format, providing a question and filters to be used in the conversation.
+    IQL prompt format, providing a question and methods to be used in the conversation.
     """
 
     def __init__(
         self,
         *,
         question: str,
-        filters: List[ExposedFunction],
-        examples: List[FewShotExample] = None,
+        methods: List[ExposedFunction],
+        examples: Optional[List[FewShotExample]] = None,
     ) -> None:
         """
         Constructs a new IQLGenerationPromptFormat instance.
 
         Args:
             question: Question to be asked.
-            filters: List of filters exposed by the view.
+            methods: List of filters exposed by the view.
             examples: List of examples to be injected into the conversation.
+            aggregations: List of aggregations exposed by the view.
         """
         super().__init__(examples)
         self.question = question
-        self.filters = "\n".join([str(filter) for filter in filters])
+        self.methods = "\n".join([str(condition) for condition in methods]) if methods else []
```

Comment:
So the type of `self.methods` can either by a string or (an empty) list? Doesn't sound right - it should be either one or the other.

As a side note, you don't need to create the temporary list inside join - `"\n".join(str(condition) for condition in methods)` would work just fine.